runtime/stdlib/test.go

```diff
@@ -453,6 +497,455 @@ func createScriptResult(inter *interpreter.Interpreter, returnValue interpreter.
+			inter,
+			interpreter.ReturnEmptyLocationRange,
+			transactionSignersFieldName,
+		)
+
+		signerAccounts := accountsFromValue(inter, signersValue)
+
+		// Get arguments
+		argsValue := transactionValue.GetMember(
+			inter,
+			interpreter.ReturnEmptyLocationRange,
+			transactionArgsFieldName,
+		)
+		args, err := arrayValueToSlice(argsValue)
+		if err != nil {
+			panic(errors.NewUnexpectedErrorFromCause(err))
+		}
+
+		err = testFramework.AddTransaction(code.Str, authorizers, signerAccounts, args)
+		if err != nil {
+			panic(err)
+		}
+
+		return interpreter.VoidValue{}
+	},
+	emulatorBackendAddTransactionFunctionType,
+)
+
+func addressesFromValue(accountsValue interpreter.Value) []common.Address {
+	accountsArray, ok := accountsValue.(*interpreter.ArrayValue)
+	if !ok {
+		panic(errors.NewUnreachableError())
+	}
+
+	addresses := make([]common.Address, 0)
+
+	accountsArray.Iterate(nil, func(element interpreter.Value) (resume bool) {
+		address, ok := element.(interpreter.AddressValue)
+		if !ok {
+			panic(errors.NewUnreachableError())
+		}
+
+		addresses = append(addresses, common.Address(address))
+
+		return true
+	})
+
+	return addresses
+}
+
+func accountsFromValue(inter *interpreter.Interpreter, accountsValue interpreter.Value) []*interpreter.Account {
+	accountsArray, ok := accountsValue.(*interpreter.ArrayValue)
+	if !ok {
+		panic(errors.NewUnreachableError())
+	}
+
+	accounts := make([]*interpreter.Account, 0)
+
+	accountsArray.Iterate(nil, func(element interpreter.Value) (resume bool) {
+		accountValue, ok := element.(interpreter.MemberAccessibleValue)
+		if !ok {
+			panic(errors.NewUnreachableError())
+		}
+
+		// Get address
+		addressValue := accountValue.GetMember(
+			inter,
+			interpreter.ReturnEmptyLocationRange,
+			accountAddressFieldName,
+		)
+		address, ok := addressValue.(interpreter.AddressValue)
+		if !ok {
+			panic(errors.NewUnreachableError())
+		}
+
+		// Get public key
+		publicKeyVal := accountValue.GetMember(
+			inter,
+			interpreter.ReturnEmptyLocationRange,
+			sema.AccountKeyPublicKeyField,
+		)
+		publicKey := publicKeyFromValue(inter, interpreter.ReturnEmptyLocationRange, publicKeyVal)
+
+		accounts = append(accounts, &interpreter.Account{
+			Address:   common.Address(address),
+			PublicKey: publicKey,
+		})
+
+		return true
+	})
+
+	return accounts
+}
+
+func hashAlgoFromValue(inter *interpreter.Interpreter, hashAlgoField interpreter.Value) sema.HashAlgorithm {
+	hashAlgoValue, ok := hashAlgoField.(interpreter.MemberAccessibleValue)
+	if !ok {
+		panic(errors.NewUnreachableError())
+	}
+
+	rawValue := hashAlgoValue.GetMember(inter, interpreter.ReturnEmptyLocationRange, sema.EnumRawValueFieldName)
+	if rawValue == nil {
+		panic(errors.NewUnreachableError())
+	}
+
+	hashAlgoRawValue, ok := rawValue.(interpreter.UInt8Value)
+	if !ok {
+		panic(errors.NewUnreachableError())
+	}
+	return sema.HashAlgorithm(hashAlgoRawValue)
+}
+
+func publicKeyFromValue(
```

Comment:
Looks like a duplicate of `runtime.NewPublicKeyFromValue`. Maybe move to e.g. `stdlib` (?). I think in one of my optimization/refactor PRs I also moved it